addons/auth_signup/controllers/main.py

```diff
@@ -20,9 +20,13 @@ def web_login(self, *args, **kw):
         ensure_db()
         response = super().web_login(*args, **kw)
         response.qcontext.update(self.get_auth_signup_config())
-        if request.httprequest.method == 'GET' and request.session.uid and request.params.get('redirect'):
-            # Redirect if already logged in and redirect param is present
-            return request.redirect(request.params.get('redirect'))
+        if request.session.uid:
+            if request.httprequest.method == 'GET' and request.params.get('redirect'):
+                # Redirect if already logged in and redirect param is present
+                return request.redirect(request.params.get('redirect'))
+            # Add message for non-internal user account without redirect if account was just created
+            if response.location == "/web/login_successful" and kw.get("confirm_password"):
+                return self.login_successful_not_internal_user(message="Registration successful.")
```

Comment:
do you really need to give the message here ?

I feel like it can be hardcoded in the template `auth_signup.login_successful` (when portal is installed, this message is just ignored)